### PR TITLE
[Data Factory]Add additionalColumns/dataConsistency/sessionLog/isolationLevel/expiryDateTime/fileListPath in ADF public swagger

### PR DESCRIPTION
**Pull request by rickysun93**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by ArcturusZhang** [comment]
LGTM but please wait for response from ARM review team

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json
Comment (ryansbenson) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json]: Adding new properties to the response of an API is classified as a breaking change. Please make these changes in a new api-version.

https://microsoft.sharepoint.com/:w:/r/teams/azureresourcemanagerteam/_layouts/15/guestaccess.aspx?e=xEcppo&share=EWXsAQ1yx25KkyYCeeWGUgwBSBxEdUDEbHi6FZ__U8EOQw

Comment (rickysun93) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json]: We had a discussion with ARM team in August that this kind of change in ADF swagger is not considered as a breaking change. Below is quote from the email between us:

_Regarding to the “definition” of breaking change, I think ADF has a looser definition than that of ARM, and looks like it’s allowable from definition of the spec :  RestAPI_BreakingChangeGuide.docx_

_“Teams MAY define backwards compatibility as their business needs require. For example, Azure defines the addition of a new JSON field in a response to be not backwards compatible. Office 365 has a looser definition of backwards compatibility and allows JSON fields to be added to responses.”_

_Back to ADF business needs, it is an data integration service which support 80+ data stores and certain number of other compute system as well, we model these data stores as **LinkedService** (with a “type” property to identify the actual data store type), and the each piece of data in these data stores as **DataSet** (with a “type” property to identify the actual data type), while we have another model object called **Pipeline** which refer to **LinkedServices** and **DataSets** to define activities to process these data. As you can see, in this model design, whenever we add a new type of **LinkedService** and **DataSet**, it will result in adding new properties of the API, but these new properties won’t impact the main concept of our model. Moreover, if we always upgrade API version in this case, it will be very difficult for customer to figure out which API version support what data stores given we have large number of  data stores and we add this data store support very quickly. So per these business requirements,  we decided not to add major version API for new connectors._

_Regarding to the swagger review process, I would expect both ARM and Swagger review team can understand this, and we don’t need to explain these details for each single swagger review._

Comment (ryansbenson) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json]: Can you forward me the email thread regarding this? rbenson@microsoft.com

The idea that RP teams can define backwards compatibility it supposed to enable teams to be more restrictive than ARM on changes, not less. 

You'll notice the O365 reference is omitted in the ARM breaking changes doc and was replaced by Azures high level view of a breaking change:
"Anything that would violate the Principle of Least Astonishment is considered a breaking change in Azure. Below are some concrete examples of what constitutes a breaking change. In the below breaking change scenarios, the API version **must** be changed."

The O365 reference comes from the following doc which describes the all up Microsoft breaking change definition:
https://github.com/Microsoft/api-guidelines/blob/vNext/Guidelines.md#123-definition-of-a-breaking-change

When reading the Microsoft docs, Azure is the business and ADF is a component of Azure.

This isn't just a random requirement we added. If you introduce new LinkedService kinds in the same api version, the existing SDK could start failing. ie If one of your new LinkedServices is returned, it will fail to deserialize. If you introduce it in a new API version, the old SDK wouldn't see the new LinkedServices and would continue to work.

If you have an exception for this, you will have to explain this to every reviewer as this is technically a breaking change according to Azure.

**Comment by ArcturusZhang** [comment]
Hi @rickysun93 do we have any updates on this PR?

**Comment by rickysun93** [comment]
> Hi @rickysun93 do we have any updates on this PR?

We're now communicating with ARM team to figure out a final decision on the api version issue, which may cost some time.

**Comment by ArcturusZhang** [comment]
> > Hi @rickysun93
> >             
> >             Haoran Sun
> >             FTE do we have any updates on this PR?
> 
> We're now communicating with ARM team to figure out a final decision on the api version issue, which may cost some time.

Thanks for the update

**Comment by ravbhatnagar** [comment]
typically this change requires a new apiversion. But since ADF has been granted an exception in the past, signing off on this. We are still working on closing this with key stakeholders.

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**
